src/client/conn/http2.rs

```diff
@@ -355,6 +351,62 @@ where
         self
     }
 
+    /// Sets the header table size.
+    ///
+    /// This setting informs the peer of the maximum size of the header compression
+    /// table used to encode header blocks, in octets. The encoder may select any value
+    /// equal to or less than the header table size specified by the sender.
+    ///
+    /// The default value of crate `h2` is 4,096.
+    pub fn header_table_size(&mut self, size: impl Into<Option<u32>>) -> &mut Self {
+        self.h2_builder.header_table_size = size.into();
+        self
+    }
+
+    /// Enables or disables server push promises.
+    ///
+    /// This value is included in the initial SETTINGS handshake.
+    /// Setting this value to value to
+    /// false in the initial SETTINGS handshake guarantees that the remote server
+    /// will never send a push promise.
+    ///
+    /// This setting can be changed during the life of a single HTTP/2
+    /// connection by sending another settings frame updating the value.
+    ///
+    /// If not set, hyper will use a default.
+    pub fn enable_push(&mut self, enabled: impl Into<Option<bool>>) -> &mut Self {
```

Comment:
I don't think it needs an option bool. The method just shouldn't exist yet, until accessing the push promises is possible.

Reply:
Okay, let's revert enable push back to its original state. We'll add a method to set it when accessing push promises becomes possible.